src/com/company/Main.java

```diff
@@ -237,31 +234,34 @@ public static char readSign() {
         return inputText.toUpperCase().charAt(0);
     }
 
-    public static void drawField(int rowCount, int[][] tab2) {
+    public static void drawField(int boardSize, final PlayerSignType[][] board) {
 
         System.out.println("");
-        String firstLine = "     ";     // 3 spacje na początku na kolumnę numerów wierszy
+        StringBuilder firstLine = new StringBuilder("     ");     // 3 spacje na początku na kolumnę numerów wierszy
 
-        for (int i = 0; i < rowCount; i++) firstLine += (char) ('a' + i) + " ! ";
-        System.out.println(firstLine);
+        for (int i = 0; i < boardSize; i++) {
+            firstLine.append((char) ('a' + i)).append(" ! ");
+        }
+        System.out.println(firstLine.toString());
 
-        String horizontalFullLine = "   ";
-        for (int i = 0; i < rowCount * 4 + 1; i++) horizontalFullLine += "-";
+        StringBuilder horizontalFullLine = new StringBuilder("   ");
+        for (int i = 0; i < boardSize * 4 + 1; i++) {
+            horizontalFullLine.append("-");
+        }
 
-        for (int i = 0; i < rowCount; i++) {
+        for (int i = 0; i < boardSize; i++) {
             System.out.println(horizontalFullLine);
             // linia z danymi z tabeli
-            String lineWithData = " " + (i + 1) + " |";
-            for (int j = 0; j < rowCount; j++) {
-                switch (tab2[i][j]) {
-                    case ItIsX:
-                        lineWithData += " X |";
-                        break;
-                    case ItIsO:
-                        lineWithData += " O |";
+            StringBuilder lineWithData = new StringBuilder(" ");
```

Comment:
StringBuilder powinien być tworzony przed for-em inaczej cała optymalizacja bierze w łeb.

Reply:
no i jest, to jest linia dla wiersza który jest tworzony w for poniżej tej deklaracji, sprawdź proszę w najnowszej wersji, ten kawałek już przerobiłam po wysłaniu do Ciebie

Reply:
wg mnie nie można, bo w tym miejscu ta linia musi być "zerowana", ale sprawdź;)

Reply:
Do zerowania wartości nie musisz tworzyć nowego obiektu.